test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,171 @@
+}
+
+template<typename alphabet_t>
+void generate_reads(std::vector<std::vector<alphabet_t> > & reads,
+                    std::vector<alphabet_t> & ref,
+                    size_t const number_of_Reads,
+                    size_t const read_length,
+                    size_t const simulated_errors,
+                    float const probInsertion,
+                    float const probDeletion
+                   )
+{
+    for(size_t i = 0; i < number_of_Reads; ++i){
+        size_t rpos = rand() % (ref.size() - read_length- simulated_errors);
+        std::vector<alphabet_t> query_tmp{ref.begin() + rpos, ref.begin() + rpos + read_length + simulated_errors};
+        for(size_t j = 0; j < simulated_errors; ++j)
+        {
+            //Substitution
+            float prob = (float) rand()/RAND_MAX;
+            if(probInsertion + probDeletion < prob)
+            {
+                mutateSubstitution(query_tmp, simulated_errors);
+            }
+            //Insertion
+            else if(probInsertion < prob)
+            {
+                mutateInsertion(query_tmp, simulated_errors);
+            }
+            //Deletion
+            else
+            {
+                mutateDeletion(query_tmp, simulated_errors);
+            }
+        }
+        std::vector<alphabet_t>  read{query_tmp.begin(), query_tmp.begin() + read_length};
+        reads.push_back(read);
+    }
+}
+
+//============================================================================
+//  undirectional; trivial_search, single, dna4, simulated error = 1
+//============================================================================
+
+void unidirectional_search_sim_e1(benchmark::State & state)
+{
+    uint8_t const searchedErrors_tmp = state.range(0);
+    uint8_t const simulated_errors = 1;
+    uint32_t index_length = 100000;
+    int number_of_Reads = 1;
+    uint32_t read_length = 100;
+    float probInsertion = 0.18;
+    float probDeletion = 0.18;
+
+    auto ref = generate_sequence_seqan3<seqan3::dna4>(index_length, 0, 0);
+    fm_index<std::vector<seqan3::dna4>> index{ref};
+    std::vector<std::vector<seqan3::dna4> > reads;
+    generate_reads(reads, ref, number_of_Reads, read_length, simulated_errors, probInsertion, probDeletion);
+    configuration cfg =  search_cfg::max_error{search_cfg::total{searchedErrors_tmp}};
+
+    for (auto _ : state)
+    {
+        auto results = search(index, reads, cfg);
+    }
+}
+
+BENCHMARK(unidirectional_search_sim_e1)
+    ->Arg(1)
+    ->Arg(2)
+    ->Arg(3);
+
+//============================================================================
+//  undirectional; trivial_search, single, dna4, searched error = 1
+//============================================================================
+
+void unidirectional_search_e3(benchmark::State & state)
+{
+    uint8_t const searchedErrors_tmp = 3;
+    uint8_t const simulated_errors = state.range(0);
+    uint32_t index_length = 100000;
+    int number_of_Reads = 1;
```

Comment:
Can I also benchmark with 500 reads the overall measured time should be more consistent, but the benchmark no longer will do multiple iterations.

Reply:
I'd say 100